agent/src/apiclient.rs

```diff
@@ -225,10 +235,16 @@ pub(super) mod api {
     /// Apiclient binary has been volume mounted into the agent container, so agent is able to
     /// invoke `/bin apiclient` to interact with the Bottlerocket Update API.
     /// This function helps to invoke apiclient raw command.
-    pub(super) async fn invoke_apiclient(args: Vec<String>) -> Result<Output> {
+    pub(super) async fn invoke_apiclient(
+        args: Vec<String>,
+        rate_limiter: Option<&SimpleRateLimiter>,
+    ) -> Result<Output> {
         let mut attempts: i8 = 0;
         // Retry up to 5 times in case the Update API is busy; Waiting 10 seconds between each attempt.
         while attempts < MAX_ATTEMPTS {
```

Comment:
Is there a reason not to use `Retry::spawn`?

Reply:
I suppose not. Looking at this, I also wonder if we should be retrying any error with the API (e.g. what if the apiserver is being restarted).